freqtrade/optimize/__init__.py

```diff
@@ -1 +1,99 @@
-from freqtrade.optimize.default_hyperopt import DefaultHyperOpts  # noqa: F401
+import logging
+from argparse import Namespace
+from typing import Any, Dict
+
+from filelock import FileLock, Timeout
+
+from freqtrade import DependencyException, constants
+from freqtrade.configuration import Configuration
+from freqtrade.state import RunMode
+
+logger = logging.getLogger(__name__)
+
+
+def setup_configuration(args: Namespace, method: RunMode) -> Dict[str, Any]:
+    """
+    Prepare the configuration for the Hyperopt module
+    :param args: Cli args from Arguments()
+    :return: Configuration
+    """
+    configuration = Configuration(args, method)
+    config = configuration.load_config()
+
+    # Ensure we do not use Exchange credentials
+    config['exchange']['key'] = ''
+    config['exchange']['secret'] = ''
+
+    if method == RunMode.BACKTEST:
+        if config['stake_amount'] == constants.UNLIMITED_STAKE_AMOUNT:
+            raise DependencyException('stake amount could not be "%s" for backtesting' %
+                                      constants.UNLIMITED_STAKE_AMOUNT)
+
```

Comment:
Should not we change the stake_amount here to "unlimited" for backtesting (as exchange.key and exchange.secret are changed above to empty vals)? Editing the config every time for backtesting (and back for other activities) is now REALLY annoying... If so, log that the stake amount is changed for backtesting...

Reply:
well but what would you set it to? 0.1, 0.5, 1 or 20? 

Notice that the check is to *not* be unlimited (it raises an exception if it is Unlimited).